rosetta-source/src/main/rosetta/product-template-type.rosetta

```diff
@@ -23,15 +23,17 @@ import cdm.product.collateral.*
 
 import cdm.mapping.config.*
 
-type TransferableProduct:  <"A TransferableProduct is a type of Product which can be used in a SettlementCommitment for a basic cash settled trade of either an Asset with or without the addition of specific EconomicTerms.">
-    asset Asset (1..1)
-    economicTerms EconomicTerms (0..1)
+type ProductBase: <"Serves as an abstract class to specify a product using a productIdentifier.">
+    productTaxonomy ProductTaxonomy (0..*) <"Specifies the product taxonomy, which is composed of a taxonomy value and a taxonomy source.">
+    economicTerms EconomicTerms (1..1)  <"The price forming features, including payouts and provisions.">
+
+type TransferableProduct extends Asset:  <"A TransferableProduct is a type of Product which can be used in a SettlementCommitment for a basic cash settled trade of either an Asset with the addition of specific EconomicTerms.">
+    economicTerms EconomicTerms (1..1)  <"The price forming features, including payouts and provisions.">
 
 type ContractualProduct extends ProductBase: <" A class to specify the contractual products' economic terms, alongside their product identification and product taxonomy. The contractual product class is meant to be used across the pre-execution, execution and (as part of the Contract) post-execution lifecycle contexts.">
     [metadata key]
     [metadata template]
-
-    economicTerms EconomicTerms (1..1) <"The economic terms associated with a contractual product, i.e. the set of features that are price-forming.">
+    productIdentifier ProductIdentifier (0..*) <"Comprises an identifier and a source. The associated metadata key denotes the ability to associate a hash value to the ProductIdentifier instantiations for the purpose of model cross-referencing, in support of functionality such as the event effect and the lineage.">
```

Comment:
Consider renaming as just `identifier`, of the same reason as above.

See also comment re positioning back into `ProductBase`.

Reply:
P3 (though contractualProduct has gone)